siriuspy/siriuspy/devices/ids.py

```diff
@@ -514,6 +640,46 @@ def cmd_change_polarization(self, polarization, timeout=None):
         return self._wait(
             self.PARAM_PVS.POL_MON, polarization, timeout=timeout, comp='eq')
 
+    def calc_move_eta(self, pparam_goal=None, kparam_goal=None):
+        """Estimate moving time for each parameter separately."""
+        # pparameter
+        param_goal, param_val = pparam_goal, self.pparameter_mon
+        param_tol = self.pparameter_tol
+        param_vel, param_acc = self.pparameter_speed, self.pparameter_accel
+        if None not in (param_goal, param_val):
```

Comment:
yes. it could be done this way too. but I prefer not two set pparam_eta twice when the conditional is satisfied.

Reply:
There is also a context thing here... Because the variable is used in the context of the method, not inside the context of the if statement. So defining it outside makes it more intuitive when reading.

Reply:
I understand what you say but python syntax allows for perfect efficient, correct and easy-to-read code using definitions inside if expressions, whether tradicional if's ou inverted (a = b if cond else c). I think this is a matter of taste. I think we should not concentrate on this.